### PR TITLE
feat(theme-common): allow passing a string for details summary

### PR DESCRIPTION
**Pull request by pReya**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [x] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

Currently, the `<Details>` component int `docusaurus-theme-common` will only accept a `summary` prop that's a React Element. So this will work:
```jsx
<Details summary={<summary>My Headline</summary>}>Some Text</Details>
```

However since in 99% of cases, the summary prop will always be wrapped in a `<summary>`, it would be nice to make the wrapper optional and also allow a string prop. This would make the `<Details>` component more end-user friendly, optionally allowing a simple signature like this:

```jsx
<Details summary={"My Headline"}>Some Text</Details>
```

## Test Plan

I manually tested all cases for this component in the dogfooding pages. However inside of mdx, we can't pass Elements as props, so this test case can't be run inside of .mdx and needs to be tested from another React component:

```jsx
<Details summary={<summary>My Headline</summary>}>Some Text</Details>
```

These were tested from the mdx file

```jsx
<Details summary={"My Headline"}>Some Text</Details>
<Details>Some Text</Details>
```

Deploy preview: https://deploy-preview-8656--docusaurus-2.netlify.app/tests/pages/markdownPageTests/#details

### CODE REVIEW

File: packages/docusaurus-theme-common/src/components/Details/index.tsx [hunk 1]
```diff
@@ -54,6 +54,16 @@ export function Details({
...
 
+  let SummaryElement = null;
+
+  if (React.isValidElement(summary)) {
+    SummaryElement = summary;
+  } else if (typeof summary === 'string') {
+    SummaryElement = <summary>{summary}</summary>;
+  } else {
+    SummaryElement = <summary>Details</summary>;
+  }
```
Comment (Josh-Cena) [inline packages/docusaurus-theme-common/src/components/Details/index.tsx:65]: I suggest:

```suggestion
  const summaryElement = React.isValidElement(summary) ? summary : <summary>{summary ?? "Details"}</summary>;
```

(Don't apply this directly; you also need to format it and change the casing of the `SummaryElement` below. You also need the `eslint-disable` comment that was there)

Comment (pReya) [reply packages/docusaurus-theme-common/src/components/Details/index.tsx]: Good idea. Changed it accordingly.

**Comment by pReya** [comment]
@Josh-Cena Any idea why the linter job fails?

**Comment by Josh-Cena** [comment]
Fixed it :)

**Review by Josh-Cena — APPROVED** [review]
LGTM; I will wait for @slorber to merge

**Comment by slorber** [comment]
Thanks 👍

**Pull request merged**